{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting bs4\n",
      "\u001b[33m  Retrying (Retry(total=4, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<pip._vendor.requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f3eec4e20d0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/bs4/\u001b[0m\n",
      "\u001b[33m  Retrying (Retry(total=3, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<pip._vendor.requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f3eec4e2910>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/bs4/\u001b[0m\n",
      "\u001b[33m  Retrying (Retry(total=2, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<pip._vendor.requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f3eec4e2450>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/bs4/\u001b[0m\n",
      "\u001b[33m  Retrying (Retry(total=1, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<pip._vendor.requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f3eec4e2090>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/bs4/\u001b[0m\n",
      "\u001b[33m  Retrying (Retry(total=0, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<pip._vendor.requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f3eec4d3090>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/bs4/\u001b[0m\n",
      "\u001b[31m  Could not find a version that satisfies the requirement bs4 (from versions: )\u001b[0m\n",
      "\u001b[31mNo matching distribution found for bs4\u001b[0m\n",
      "\u001b[33mYou are using pip version 9.0.1, however version 10.0.1 is available.\n",
      "You should consider upgrading via the 'pip install --upgrade pip' command.\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#!pip install natsort\n",
    "!pip install bs4\n",
    "import pickle\n",
    "with open('unraveled_links.p','rb') as handle:\n",
    "    winners = pickle.load(handle)\n",
    "\n",
    "from bs4 import BeautifulSoup\n",
    "def html_to_txt(content):\n",
    "    soup = BeautifulSoup(content, 'html.parser')\n",
    "    #strip HTML\n",
    "    for script in soup([\"script\", \"style\"]):\n",
    "        script.extract()    # rip it out\n",
    "    text = soup.get_text()\n",
    "    #organize text\n",
    "    lines = (line.strip() for line in text.splitlines())  # break into lines and remove leading and trailing space on each\n",
    "    chunks = (phrase.strip() for line in lines for phrase in line.split(\"  \")) # break multi-headlines into a line each\n",
    "    text = '\\n'.join(chunk for chunk in chunks if chunk) # drop blank lines\n",
    "    str_text = str(text)\n",
    "    \n",
    "    return str_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/jovyan/Examples/rcgerkinscholar32.html\n",
      "/home/jovyan/Examples/rcgerkinscholar23.html\n",
      "{   'cliau': 13.96,\n",
      "    'english': True,\n",
      "    'eofh': 8.82767382403349,\n",
      "    'fkg': 8.1,\n",
      "    'fre': 58.89,\n",
      "    'frequencies': None,\n",
      "    'frexMost': None,\n",
      "    'gf': 11.526486872018902,\n",
      "    'info_density': 0.0001628896626147863,\n",
      "    'link': '/home/jovyan/Examples/rcgerkinscholar23.html',\n",
      "    'penalty': 23.139790421937715,\n",
      "    'scaled_info_density': 7.240184132580065e-09,\n",
      "    'smog': 11.3,\n",
      "    'sp': 0.08690195077459181,\n",
      "    'ss': 0.4100713094557451,\n",
      "    'standard': '11th and 12th grade',\n",
      "    'uniqueness': 7.390753291959934e-06,\n",
      "    'wcount': 22498}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "{   'cliau': 13.92,\n",
      "    'english': True,\n",
      "    'eofh': 9.969227657832072,\n",
      "    'fkg': 11.4,\n",
      "    'fre': 44.95,\n",
      "    'frequencies': None,\n",
      "    'frexMost': None,\n",
      "    'gf': 12.242697965021055,\n",
      "    'info_density': 0.010121377456213181,\n",
      "    'link': '/home/jovyan/Examples/rcgerkinscholar32.html',\n",
      "    'penalty': 24.525981423600392,\n",
      "    'scaled_info_density': 8.009446581581714e-08,\n",
      "    'smog': 13.3,\n",
      "    'sp': 0.040594090989445004,\n",
      "    'ss': 0.4101277770383565,\n",
      "    'standard': '12th and 13th grade',\n",
      "    'uniqueness': 6.22155249645671e-07,\n",
      "    'wcount': 126368}\n"
     ]
    }
   ],
   "source": [
    "def print_best_text(fileName):\n",
    "    file = open(fileName)\n",
    "    if str('.html') in fileName:\n",
    "        text = html_to_txt(file)\n",
    "    else:\n",
    "        text = convert_pdf_to_txt(file)\n",
    "    file.close()\n",
    "    return text\n",
    "\n",
    "print(winners[1]['link'])\n",
    "print(winners[0]['link'])#)\n",
    "\n",
    "import pprint\n",
    "pp = pprint.PrettyPrinter(indent=4)\n",
    "pp.pprint(winners[0])\n",
    "print('\\n\\n\\n')\n",
    "#pp.pprint(text0)\n",
    "pp.pprint(winners[1])\n",
    "#pp.pprint(text1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: 'Neutronyahoo66.html'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-3-1035b4f03bde>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mpp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprint_best_text\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'Neutronyahoo66.html'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m<ipython-input-2-39b1127209f5>\u001b[0m in \u001b[0;36mprint_best_text\u001b[0;34m(fileName)\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mprint_best_text\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfileName\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m     \u001b[0mfile\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfileName\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'.html'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mfileName\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m         \u001b[0mtext\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhtml_to_txt\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfile\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: 'Neutronyahoo66.html'"
     ]
    }
   ],
   "source": [
    "\n",
    "pp.pprint(print_best_text('Neutronyahoo66.html'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('Transgenic solutions for the germline\\n'\n",
      " 'Embryo series courtesy of Einhard Schierenberg\\n'\n",
      " 'Home\\n'\n",
      " 'About\\n'\n",
      " 'Contact\\n'\n",
      " 'Discuss WormBook\\n'\n",
      " 'Sign up for e-alerts\\n'\n",
      " 'Send feedback\\n'\n",
      " 'Cite WormBook\\n'\n",
      " 'for Authors\\n'\n",
      " 'Downloads\\n'\n",
      " 'Download all chapters\\n'\n",
      " 'Import EndNote citations\\n'\n",
      " 'Import RefMan citations\\n'\n",
      " 'Sponsors\\n'\n",
      " 'Links\\n'\n",
      " 'WormBase\\n'\n",
      " 'WormAtlas\\n'\n",
      " 'Textpresso\\n'\n",
      " 'HTML\\n'\n",
      " 'PDF\\n'\n",
      " 'or Search with Textpresso\\n'\n",
      " 'Download PDF version\\n'\n",
      " 'Transgenic solutions for the germline*\\n'\n",
      " 'Christopher Merritt, Christopher M. Gallo, Dominique Rasoloson, and '\n",
      " 'Geraldine Seydoux§Department of Molecular Biology and Genetics, Howard '\n",
      " 'Hughes Medical Institute, and Center for Cell Dynamics, Johns Hopkins\\n'\n",
      " 'School of Medicine; 725 N. Wolfe St., PCTB 706, Baltimore, MD 21205, USA\\n'\n",
      " 'This chapter is in WormBook section:\\n'\n",
      " '> WormMethods\\n'\n",
      " 'View/Add Comments\\n'\n",
      " 'Table of Contents\\n'\n",
      " '1. Introduction\\n'\n",
      " '2. The germline challenge\\n'\n",
      " '3. Transgene construction by modular cloning: the Multisite Gateway® System\\n'\n",
      " '4. Considerations when designing entry clones.\\n'\n",
      " '4.1. Choosing the promoter\\n'\n",
      " '4.2. Choosing the ORF\\n'\n",
      " \"4.3. Choosing the 3'UTR\\n\"\n",
      " '4.4. Consideration when assembling a transformation-ready transgene\\n'\n",
      " '4.5. Where to put the start and stop codon?\\n'\n",
      " '4.6. Other linker issue\\n'\n",
      " '5. Choosing a destination vector\\n'\n",
      " '6. Non-Multisite Gateway® vectors\\n'\n",
      " '6.1. Traditional cloning vectors\\n'\n",
      " '6.2. Single site Gateway® vectors\\n'\n",
      " '7. Protocols\\n'\n",
      " '7.1. Gateway® cloning\\n'\n",
      " '7.2. Oligo design/PCR\\n'\n",
      " '7.3. BP Reaction\\n'\n",
      " '7.4. BP Reaction Transformation\\n'\n",
      " '7.5. Multisite LR Reaction\\n'\n",
      " '7.6. Multisite LR Reaction Transformation\\n'\n",
      " '7.7. Bombardment Protocol\\n'\n",
      " '8. References\\n'\n",
      " '1.\\xa0Introduction\\n'\n",
      " 'One of the most thrilling experiments in biology is to introduce a gene of '\n",
      " \"one's own design into a favorite animal and examine\\n\"\n",
      " 'the effect in the transgenic progeny. Methods to construct, transform and '\n",
      " 'monitor transgenes have been available to worm breeders\\n'\n",
      " 'since the pioneering work of Andy Fire and Craig Mello (Fire, 1986; Mello '\n",
      " 'and Fire, 1995) and the introduction of green fluorescent protein (GFP) by '\n",
      " 'Marty Chalfie (Chalfie et al., 1994). Sadly, for many years, the thrill of '\n",
      " '“seeing green” was denied to worm breeders working on the germline, as '\n",
      " 'transgenes\\n'\n",
      " 'stubbornly refused to express in germ cells. In 1997, Bill Kelly and Andy '\n",
      " 'Fire showed that transgene silencing in the germline\\n'\n",
      " 'is a copy-number driven process (Kelly et al., 1997). Multi-copy transgenes '\n",
      " 'are expressed in the soma but silenced in the germline; in contrast, '\n",
      " 'low-copy transgenes are expressed\\n'\n",
      " 'in both. Today, new transformation methods make it possible to routinely '\n",
      " 'obtain low copy transgenes inserted in the genome.\\n'\n",
      " 'In this chapter, we review these methods and give practical advice for '\n",
      " 'designing and transforming “germline-ready” transgenes.\\n'\n",
      " '2.\\xa0The germline challenge\\n'\n",
      " 'The experiments of Kelly and Fire showed that unlike somatic cells, germ '\n",
      " 'cells are very efficient at silencing genes present\\n'\n",
      " 'in multi-copy. In some cases, multi-copy transgenes have also been observed '\n",
      " 'to silence the corresponding endogenous locus\\n'\n",
      " 'and phenocopy a loss-of-function mutation (co-suppression) (Robert et al., '\n",
      " '2004). Silencing affects transgene expression not only in all germ cells, '\n",
      " 'but also in the blastomeres of early embryos, which\\n'\n",
      " 'depend primarily on maternal mRNAs and proteins synthesized during '\n",
      " 'oogenesis.\\n'\n",
      " 'Three methods have been described to obtain low-copy transgenes. The first, '\n",
      " 'developed by Kelly and Fire, involve diluting\\n'\n",
      " 'the transgene with genomic DNA prior to injection to make a “complex array” '\n",
      " '(Kelly et al., 1997). Complex arrays, however, are difficult to obtain and '\n",
      " 'prone to silencing over time. Two other methods yield transgenes that\\n'\n",
      " 'are inserted in the genome, either at random sites [gene-gun transformation '\n",
      " '(Praitis et al., 2001; Wilm et al., 1999)] or at a pre-selected site '\n",
      " '[Mos1-mediated Single Copy Insertion (MosSCI) (Frokjaer-Jensen et al., '\n",
      " '2008)]. These methods yield stable lines that can express in the germline '\n",
      " 'for many generations.\\n'\n",
      " 'Besides silencing, gene expression in the germline presents two other unique '\n",
      " 'challenges. First, unlike in the soma where promoters\\n'\n",
      " \"do the lion's share of gene regulation, in the germline, 3'UTRs often (but \"\n",
      " 'not always) specify expression patterns (Merritt et al., 2008). Second, most '\n",
      " 'of germline development occurs in a syncytium, where germ cells in different '\n",
      " 'developmental stages are connected\\n'\n",
      " 'to a common central cytoplasm (rachis) (Hubbard and Greenstein, 2005). '\n",
      " 'Diffusion into the rachis means that transgenic proteins, in particular '\n",
      " 'small reporters like GFP, can end up far from the\\n'\n",
      " 'site of synthesis (Figure 1).\\n'\n",
      " 'Figure 1: The choice of the ORF can affect the final observed pattern.\\xa0'\n",
      " 'Comparison of three types of GFP fusions. Each fusion is driven by the pie-1 '\n",
      " \"promoter and the indicated 3'UTR (mex-5 or fbf-2). GFP:ORF is an \"\n",
      " 'amino-terminal fusion between GFP and the indicated ORF (mex-5 or fbf-2) '\n",
      " '(Merritt et al., 2008). GFP:H2B is an amino-terminal fusion between GFP and '\n",
      " 'histone H2B. GFP is GFP alone. The GFP alone fusion gives the least\\n'\n",
      " 'accurate/specific pattern, due to diffusion through the rachis (C.M., '\n",
      " \"unpublished). The GFP:H2B pattern reflects regulation by the 3'UTR only. The \"\n",
      " 'GFP:ORF pattern gives a pattern most similar to the endogenous\\n'\n",
      " 'pattern. Inclusion of the ORF is particularly important for the proper '\n",
      " 'distribution of maternal proteins in embryos, which\\n'\n",
      " 'often depend on ORF sequences for segregation during early cleavages.\\n'\n",
      " 'For these reasons, when designing germline transgenes, all components '\n",
      " \"(promoter, tag, ORF and 3'UTR) must be considered carefully.\\n\"\n",
      " 'Short cuts commonly used in somatic tissues, such as relying on simple '\n",
      " \"promoter fusions (promoter::GFP::unc-54 3'UTR) to determine expression \"\n",
      " 'patterns (Hunt-Newbury et al., 2007), cannot be used in the germline. The '\n",
      " 'methods presented in this chapter are designed to facilitate the cloning and '\n",
      " 'mix-and-matching\\n'\n",
      " 'of different gene sequences to give the Worm Breeder maximum control over '\n",
      " 'transgene design. The methods here are also compatible\\n'\n",
      " 'with the use of clones from publicly available collections, such as the '\n",
      " 'promoterome (Dupuy et al., 2004), the ORFeome (Lamesch et al., 2004), and '\n",
      " \"3'UTRome (Mangone et al., 2008).\\n\"\n",
      " '3.\\xa0Transgene construction by modular cloning: the Multisite Gateway® '\n",
      " 'System\\n'\n",
      " 'Modular cloning replaces traditional restriction-enzyme cloning with '\n",
      " 'recombinational cloning, which allows users to rapidly\\n'\n",
      " \"recombine different parts (promoter, ORF and 3'UTR) of the transgene at \"\n",
      " 'will. Modular cloning is becoming more and more popular\\n'\n",
      " \"as collections of systematically-cloned promoters, ORFs, and 3'UTRs are \"\n",
      " 'becoming available. The method we describe in detail\\n'\n",
      " 'here is based on the Multisite Gateway® System from Invitrogen. The '\n",
      " 'Multisite Gateway® system allows for rapid mix-and-match cloning of up to '\n",
      " 'three DNA fragments in two steps:\\n'\n",
      " 'Step 1: Construction of Entry Clones\\n'\n",
      " \"Each element (i.e. promoter, ORF and 3' sequence) is cloned into a Donor \"\n",
      " 'Vector to create an Entry clone (Figure 2). There are three Donor Vectors, '\n",
      " \"each corresponding to the specific position (5', middle, or 3') that the \"\n",
      " 'insert will occupy\\n'\n",
      " 'in the final clone (transformation-ready transgene). Typically promoters are '\n",
      " \"cloned in the 5' position, ORFs in the middle,\\n\"\n",
      " \"and 3'UTRs in the 3' position. Cloning is done by:\\n\"\n",
      " 'PCR amplification of the desired fragment using oligos that contain '\n",
      " 'sequences complementary to the insert as well as sequences\\n'\n",
      " 'for cloning by recombination\\n'\n",
      " 'recombination cloning (BP reaction) into one of the three Gateway® Donor '\n",
      " 'Vectors:\\n'\n",
      " \"5' POSITION: pDONRP4-P1R, MIDDLE POSITION: pDONR201/221 (either can be \"\n",
      " 'used),\\n'\n",
      " \"3' POSITION: pDONRP2-P3\\n\"\n",
      " 'sequence verification of the resulting Entry clone to ensure that no '\n",
      " 'mistakes were introduced during the PCR amplification.\\n'\n",
      " 'Figure 2: Step 1: Assembly of Entry Clone (BP reaction).\\xa0The desired '\n",
      " 'fragment is PCR amplified using oligos that contain attB sequences. '\n",
      " 'Recombination with a Gateway® Donor Vector gives rise to an Entry clone.\\n'\n",
      " 'Step 2: Assembly of transformation-ready transgene\\n'\n",
      " 'Three Entry clones are recombined in a single cloning reaction (LR reaction) '\n",
      " 'with the Destination vector (Figure 3). The Destination vector contains the '\n",
      " 'marker(s) used for transformation into worms (for example: pCG150, which '\n",
      " 'contains the\\n'\n",
      " 'unc-119 rescuing fragment). The resulting final construct is the transgene '\n",
      " 'ready for transformation.\\n'\n",
      " 'Figure 3: Step 2: Assembly of transformation-ready transgene (LR '\n",
      " 'reaction).\\xa0Three Entry clones are recombined in a single reaction with '\n",
      " 'the Destination vector to give rise to the final construct '\n",
      " '(transformation-ready\\n'\n",
      " 'transgene). The Destination vector contains the marker used for '\n",
      " 'transformation into worms (unc-119 in this example).\\n'\n",
      " '4.\\xa0Considerations when designing entry clones.\\n'\n",
      " 'For general instructions on how to design entry clones, please consult the '\n",
      " 'Multisite Gateway® Technology (Invitrogen) website '\n",
      " '(http://www.invitrogen.com/site/us/en/home/Products-and-Services/Applications/Cloning/Gateway-Cloning/MultiSite '\n",
      " '-Gateway-Technology.html) and manual (note: DO NOT use the Multisite '\n",
      " 'Gateway® “PRO” manual). See the Protocols section in this chapter for '\n",
      " 'additional instructions/tips.\\n'\n",
      " '4.1.\\xa0Choosing the promoter\\n'\n",
      " '- Promoter of gene of interest: this choice is recommended when the goal is '\n",
      " 'to design a transgene that will most closely approximate the expression '\n",
      " 'pattern\\n'\n",
      " 'of the gene of interest (in which case all parts of the gene (promoter, ORF '\n",
      " \"and 3'UTR) must be used). Typically, the intergenic\\n\"\n",
      " 'sequence from the end/start of the upstream ORF to the start codon of the '\n",
      " 'gene of interest is sufficient. Large introns can\\n'\n",
      " 'also be included.\\n'\n",
      " '- pie-1 promoter: this promoter is suitable for expression in all germ cells '\n",
      " 'from the late L1 stage to the adult stage (Merritt et al., 2008). When '\n",
      " \"combined with GFP:Histone H2B and tubulin 3'UTR, expression is first \"\n",
      " 'detected in the late L1 stage and is maintained\\n'\n",
      " 'in all germ cells through spermatogenesis and oogenesis. Maternal expression '\n",
      " 'persists in embryos with declining levels after\\n'\n",
      " 'the 100-cell stage. More restricted expression can be obtained by combining '\n",
      " \"the pie-1 promoter with specific 3'UTRs, see below.\\n\"\n",
      " '- spe-11 promoter: this promoter is active specifically during '\n",
      " 'spermatogenesis in both hermaphrodites and males (Merritt et al., 2008). '\n",
      " \"When combined with GFP:Histone H2B and tubulin 3'UTR, expression is first \"\n",
      " 'detected in late pachytene and is maintained\\n'\n",
      " 'through spermatogenesis.\\n'\n",
      " '- hsp promoter: for heat-inducible expression. CAUTION: These promoters are '\n",
      " 'not germline-specific, they express in most somatic cells. In\\n'\n",
      " 'the germline, expression is first seen weakly in late pachytene of '\n",
      " 'spermatogenesis and oogenesis (C. M. unpublished) several\\n'\n",
      " 'hours (usually, > 4 hrs) after heat-shock.\\n'\n",
      " \"(See Table 1A for some useful 5' Entry Clones containing promoters.)\\n\"\n",
      " 'Table 1: Selected Entry Clones\\n'\n",
      " 'A. 5’ Entry Clones\\n'\n",
      " 'Element\\n'\n",
      " 'Application\\n'\n",
      " '(to drive expression)\\n'\n",
      " 'Description\\n'\n",
      " 'Entry Clone\\n'\n",
      " '(reference)\\n'\n",
      " 'Distributed by\\n'\n",
      " 'pie-1 promoter\\n'\n",
      " 'throughout the germline\\n'\n",
      " 'Contains promoter + large intron from pie-1 locus (3.0kb total in length)\\n'\n",
      " 'pCG142 (Merritt et al., 2008)\\n'\n",
      " 'Addgene (http://www.addgene.org/pgvec1)\\n'\n",
      " 'pie-1 promoter (short)\\n'\n",
      " 'throughout the germline\\n'\n",
      " 'smaller pie-1 promoter (1.1 kb) lacking repetitive sequences (which are '\n",
      " 'thought to be problematic with MosSCI)\\n'\n",
      " 'pCM1.58 (C.M. unpublished)\\n'\n",
      " 'Addgene (http://www.addgene.org/pgvec1)\\n'\n",
      " 'pie-1 promoter (short, no ATG)\\n'\n",
      " 'throughout the germline\\n'\n",
      " 'smaller pie-1 promoter (1.1 kb) with no start codon. Avoids attB1-encoded '\n",
      " 'amino-acids in fusion protein. ATG must be included in middle\\n'\n",
      " 'cassette (see Figure 4C).\\n'\n",
      " 'pCM1.127 (C.M. unpublished)\\n'\n",
      " 'Addgene (http://www.addgene.org/pgvec1)\\n'\n",
      " 'spe-11 promoter\\n'\n",
      " 'in sperm\\n'\n",
      " '272 bp spe-11 promoter.\\n'\n",
      " 'pCM1.41 (Merritt et al., 2008)\\n'\n",
      " 'Addgene (http://www.addgene.org/pgvec1)\\n'\n",
      " 'spe-11 promoter (no ATG)\\n'\n",
      " 'in sperm\\n'\n",
      " '272 bp spe-11 promoter with no start codon. Avoids attB1-encoded AAs in '\n",
      " 'final fusion protein. ATG must be included in middle cassette (see\\n'\n",
      " 'Figure 4C).\\n'\n",
      " 'pCM1.128 (C.M. unpublished)\\n'\n",
      " 'Addgene (http://www.addgene.org/pgvec1)\\n'\n",
      " 'hsp16-2/hsp16-41\\n'\n",
      " 'heat-inducible\\n'\n",
      " 'heatshock promoters commonly used for heat-inducible expression. Cloned from '\n",
      " 'vectors created by Andy Fire.\\n'\n",
      " 'pCM1.56/1.57 (C.M. unpublished)\\n'\n",
      " 'Addgene (http://www.addgene.org/pgvec1)\\n'\n",
      " 'Other germline promoters (Seydoux lab)\\n'\n",
      " 'throughout the germline\\n'\n",
      " 'many other germline promoters ranging in length from 400bp to 3000bp (see '\n",
      " 'Merritt et al., 2008 - Table S5).\\n'\n",
      " 'various (Merritt et al., 2008)\\n'\n",
      " 'Addgene (http://www.addgene.org/pgvec1)\\n'\n",
      " 'C. elegans Promoterome library (Vidal lab)\\n'\n",
      " 'specific (somatic) cell-types\\n'\n",
      " 'These clones contain up to 2 kb of intergenic sequence upstream of the ATG. '\n",
      " 'They contain ATGs and are cloned as shown in Figure 4B.\\n'\n",
      " 'various (Dupuy et al., 2004)\\n'\n",
      " 'Geneservice '\n",
      " '(http://www.geneservice.co.uk/products/clones/Celegans_Prom.jsp)\\n'\n",
      " 'B. Middle Entry Clones\\n'\n",
      " 'Element\\n'\n",
      " 'Application\\n'\n",
      " 'Description\\n'\n",
      " 'Entry Clone (REF)\\n'\n",
      " 'Distributed by\\n'\n",
      " 'GFP\\n'\n",
      " 'tag ORF with GFP\\n'\n",
      " 'GFP with 3 synthetic introns (from Andy Fire)\\n'\n",
      " 'pCM1.53 (Merritt et al., 2008)\\n'\n",
      " 'Addgene (http://www.addgene.org/pgvec1)\\n'\n",
      " 'mCherry\\n'\n",
      " 'tag ORF with mCherry\\n'\n",
      " 'mCherry with C. elegans codon preferences and 3 synthetic introns (created '\n",
      " 'by Chris Gallo; Seydoux lab)\\n'\n",
      " 'pCG144 (Merritt et al., 2008)\\n'\n",
      " 'Addgene (http://www.addgene.org/pgvec1)\\n'\n",
      " 'GFP:histone H2B\\n'\n",
      " 'reporter\\n'\n",
      " 'pCM1.35 (Merritt et al., 2008)\\n'\n",
      " 'Addgene (http://www.addgene.org/pgvec1)\\n'\n",
      " 'mCherry:histone H2B\\n'\n",
      " 'reporter\\n'\n",
      " 'pCM1.151 (Merritt et al., 2008)\\n'\n",
      " 'Addgene (http://www.addgene.org/pgvec1)\\n'\n",
      " 'C. elegans ORFeome library (Vidal lab)\\n'\n",
      " 'express ORF\\n'\n",
      " 'these ORFs do not contain start or stop codons and thus must be added in '\n",
      " 'flanking cassettes\\n'\n",
      " 'various (http://worfdb .dfci.harvard.edu; Dupuy et al., 2004)\\n'\n",
      " 'Geneservice (http://www.geneservice.co.uk/products/cdna/Celegans_ORF.jsp)\\n'\n",
      " \"C. 3' Entry Clones\\n\"\n",
      " 'Element\\n'\n",
      " 'Application (to drive expression)\\n'\n",
      " 'Description\\n'\n",
      " 'Entry Clone (REF)\\n'\n",
      " 'Distributed by\\n'\n",
      " \"tbb-2 3'UTR\\n\"\n",
      " 'in all cell types\\n'\n",
      " \"3'UTR permissive for expression throughout the germline\\n\"\n",
      " 'pCM1.36 (Merritt et al., 2008)\\n'\n",
      " 'Addgene (http://www.addgene.org/pgvec1)\\n'\n",
      " \"pie-1 3'UTR\\n\"\n",
      " 'in oocytes and embryos\\n'\n",
      " \"3'UTR with highest expression in oocytes and early embryos\\n\"\n",
      " 'pCM5.47 (Merritt et al., 2008)\\n'\n",
      " 'Addgene (http://www.addgene.org/pgvec1)\\n'\n",
      " \"unc-54 3'UTR\\n\"\n",
      " 'in all cell types\\n'\n",
      " \"3'UTR permissive for expression in all cells (Fire vectors)\\n\"\n",
      " 'pCM5.37 (Merritt et al., 2008)\\n'\n",
      " 'Addgene (http://www.addgene.org/pgvec1)\\n'\n",
      " 'C. elegans UTRome library (Piano lab)\\n'\n",
      " 'to drive expression in various cell types\\n'\n",
      " \"Last 30 bases of ORF sequence + STOP codon + 3'UTR sequence to polyA\\n\"\n",
      " 'various (UTRome.org; Mangone et al., 2008))\\n'\n",
      " 'not yet determined\\n'\n",
      " \"Other germline 3'UTRs (Seydoux lab)\\n\"\n",
      " 'to drive expression in different germline cell types\\n'\n",
      " 'various (see Table 2) (Merritt et al., 2008)\\n'\n",
      " 'Addgene (http://www.addgene.org/pgvec1)\\n'\n",
      " '4.2.\\xa0Choosing the ORF\\n'\n",
      " '- ORF of gene of interest: this choice is recommended when the goal is to '\n",
      " 'design a transgene that will most closely approximate the expression '\n",
      " 'pattern\\n'\n",
      " 'of the gene of interest. Different proteins will have different turnover '\n",
      " 'rates and diffusion patterns in the germline.\\n'\n",
      " '- GFP, mCherry, and other fluorescent proteins can be used in germ cells as '\n",
      " 'has been described for somatic transgenes (see Reporter gene fusions). Note '\n",
      " 'that since most germ cells are connected to each other through the rachis '\n",
      " '(with the exception of developing sperm),\\n'\n",
      " 'small proteins such as GFP can diffuse from cell to cell through most of the '\n",
      " 'gonad (Figure 1).\\n'\n",
      " '- GFP:H2B (histone H2B) localizes efficiently to chromatin. This reporter is '\n",
      " 'a good choice when the goal is to 1) concentrate the GFP\\n'\n",
      " 'signal to a small area to boost detection, 2) limit protein diffusion '\n",
      " 'through the rachis, and 3) test the ability of a promoter\\n'\n",
      " \"or 3'UTR to restrict expression to specific stages (Merritt et al., 2008). \"\n",
      " 'Note that the rate of H2B turnover during germline development is not known '\n",
      " 'and could in theory artificially extend or\\n'\n",
      " 'shorten an expression pattern.\\n'\n",
      " \"(See Table 1B for some useful 5' Entry Clones containing ORFs)\\n\"\n",
      " \"4.3.\\xa0Choosing the 3'UTR\\n\"\n",
      " \"- 3'UTR of gene of interest: Many genes in the germline are regulated \"\n",
      " \"post-transcriptionally using sequences in the 3'UTRs (Merritt et al., 2008). \"\n",
      " \"Using the 3'UTR of the gene of interest is ESSENTIAL when the goal is to \"\n",
      " 'design a transgene that will most closely approximate\\n'\n",
      " 'the expression pattern of the gene of interest. See Table 1C and Table 2 for '\n",
      " \"3'UTRs with defined expression patterns.\\n\"\n",
      " \"- tbb-2 and unc-54 3'UTRs are compatible with expression in all germ cells \"\n",
      " \"(Merritt et al., 2008). See Tables 1C and 2 for some useful 3' Entry Clones \"\n",
      " \"containing 3'UTRs.\\n\"\n",
      " \"Table 2: 3'UTRs with different expression patterns in the germline\\n\"\n",
      " 'progenitors\\n'\n",
      " 'pachytene\\n'\n",
      " 'loop\\n'\n",
      " 'oocytes\\n'\n",
      " 'embryos (C.M. unpublished)\\n'\n",
      " \"3'UTR\\n\"\n",
      " 'Entry\\n'\n",
      " 'distal\\n'\n",
      " 'proximal\\n'\n",
      " 'distal\\n'\n",
      " 'proximal\\n'\n",
      " 'distal\\n'\n",
      " 'medial\\n'\n",
      " 'proximal\\n'\n",
      " 'early\\n'\n",
      " 'late\\n'\n",
      " \"Germline expression of GFP:H2B when driven by the given 3'UTR (Merritt et \"\n",
      " 'al., 2008). Gonadal regions are listed distally to proximally, from left to '\n",
      " 'right. ++ indicates strongest domain(s) of GFP:H2B expression,\\n'\n",
      " '+ indicates weaker domain(s) of GFP expression, and - indicates no GFP '\n",
      " \"expression. Other 3'UTRs that provide similar expression\\n\"\n",
      " 'patterns: 1mes-2, par-5, fog-2, spe-11, spe-38, spe-41, 2fbf-2, 3lip-1, and '\n",
      " '4mes-3.\\n'\n",
      " 'tbb-21\\n'\n",
      " 'pCM1.36\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " 'pgl-1\\n'\n",
      " 'pCM5.30\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " 'nos-3\\n'\n",
      " 'pCM5.35\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " 'glp-1\\n'\n",
      " 'pCM5.40\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '++\\n'\n",
      " 'fbf-12\\n'\n",
      " 'pCM5.33\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '−\\n'\n",
      " '−\\n'\n",
      " '−\\n'\n",
      " '−\\n'\n",
      " '+ (posterior)\\n'\n",
      " 'daz-1\\n'\n",
      " 'pCM5.43\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " 'pgl-3\\n'\n",
      " 'pCM5.46\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " 'gld-1\\n'\n",
      " 'pCM5.36\\n'\n",
      " '−\\n'\n",
      " '+\\n'\n",
      " '++\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '−\\n'\n",
      " '+ (posterior)\\n'\n",
      " 'him-3\\n'\n",
      " 'pCM5.52\\n'\n",
      " '−\\n'\n",
      " '−\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " 'rme-23\\n'\n",
      " 'pCM5.51\\n'\n",
      " '−\\n'\n",
      " '−\\n'\n",
      " '+\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " 'mex-5\\n'\n",
      " 'pCM5.48\\n'\n",
      " '−\\n'\n",
      " '−\\n'\n",
      " '−\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " 'puf-5\\n'\n",
      " 'pCM5.64\\n'\n",
      " '−\\n'\n",
      " '−\\n'\n",
      " '−\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " 'pal-1\\n'\n",
      " 'pCM5.49\\n'\n",
      " '−\\n'\n",
      " '−\\n'\n",
      " '−\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '+\\n'\n",
      " '++\\n'\n",
      " 'spn-4\\n'\n",
      " 'pCM5.50\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '−\\n'\n",
      " '−\\n'\n",
      " '−\\n'\n",
      " '−\\n'\n",
      " '−\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " 'pie-1\\n'\n",
      " 'pCM5.47\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " 'pos-1\\n'\n",
      " 'pCM5.66\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " 'cye-14\\n'\n",
      " 'pCM5.54\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " 'cep-1\\n'\n",
      " 'pCM5.44\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " 'mex-3\\n'\n",
      " 'pCM5.45\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '−\\n'\n",
      " '−\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " '++\\n'\n",
      " 'fog-1\\n'\n",
      " 'pCM5.60\\n'\n",
      " '−\\n'\n",
      " '++\\n'\n",
      " '+\\n'\n",
      " '++\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '+\\n'\n",
      " '4.4.\\xa0Consideration when assembling a transformation-ready transgene\\n'\n",
      " 'The main advantage of Gateway® cloning is the ability to use pre-made Entry '\n",
      " 'clones and thus reduce the number of clones that must be made de novo by the '\n",
      " 'user. In general, the design of a transformation-ready transgene will be '\n",
      " 'dictated by whatever pre-made Entry clones\\n'\n",
      " 'are available and the purpose of the transgene. For example, if the goal is '\n",
      " 'simply to tag a protein under endogenous control,\\n'\n",
      " 'one could consider cloning the promoter in the first position, the tag in '\n",
      " \"the second position, and the ORF + 3'UTR in the\\n\"\n",
      " 'third position (amplified as a single unit from genomic DNA or cDNA) (Figure '\n",
      " '4B). In this way, the user can take advantage of several pre-made middle '\n",
      " 'Entry clones containing different tags (Table 1B) to rapidly create and test '\n",
      " 'different tagged versions of the protein of interest.\\n'\n",
      " 'Figure 4: attB sites in final constructs.\\n'\n",
      " 'A. The frame usage of attB sites must be considered for final constructs. '\n",
      " 'These reading frames are suggested by Invitrogen and are used for all '\n",
      " 'clones\\n'\n",
      " 'listed in Table 1. Extra nucleotides flanking the attB sites (in red) must '\n",
      " 'be added for proper reading frame usage.\\n'\n",
      " \"B. Example of a construct with the start codon in the 5' cassette and a stop \"\n",
      " \"codon in the 3' cassette. In the 5' cassette\\n\"\n",
      " '(containing a promoter) an A (in red) is added after the ATG so that the '\n",
      " 'attB1 sequences will be in frame. In the middle cassette (containing GFP) CA '\n",
      " \"(in red) is added 5' of the GFP coding sequence\\n\"\n",
      " 'and an A (in red) is added after the final GFP codon to maintain frame '\n",
      " \"through the attB2 site. In the 3' cassette (containing\\n\"\n",
      " \"ORF::3'UTR) GA (in red) is added before the ORF coding sequence to maintain \"\n",
      " 'frame through the ORF. In this strategy, an available\\n'\n",
      " \"promoter can be used in the 5' cassette (Table 1A) and an available \"\n",
      " 'fluorescent tag can be used in the middle cassette (Table\\n'\n",
      " \"1B). The 3' cassette containing ORF::3'UTR is the only entry clone that \"\n",
      " 'needs to be created by the researcher. We typically\\n'\n",
      " \"clone the ORF::3'UTR (with introns) directly from a genomic DNA template, \"\n",
      " 'but a cDNA template can also be used.\\n'\n",
      " 'C. Example of a construct with the start codon and stop codon in the middle '\n",
      " 'cassette. This strategy avoids the inclusion of\\n'\n",
      " 'attB sequences in the final ORF. Available promoters (lacking ATG) (Table '\n",
      " \"1A) and 3'UTRs (Table 1C) can be used in the 5'\\n\"\n",
      " \"and 3' cassettes. The middle cassette must contain the ORF and fluorescent \"\n",
      " 'tag fused by fusion PCR.\\n'\n",
      " '4.5.\\xa0Where to put the start and stop codon?\\n'\n",
      " 'Placement of the start and stop codons will determine whether the final '\n",
      " 'protein fusion will contain amino acids coded by the\\n'\n",
      " 'att recombination sites used in cloning. Ideally the start and stop codons '\n",
      " 'should flank the ORF directly to avoid adding any\\n'\n",
      " 'extensions to the transgenic protein (Figure 4C). In practice though, if '\n",
      " 'using clones from available libraries, you may be forced to consider a '\n",
      " 'different strategy.\\n'\n",
      " '4.5.1.\\xa0- ATG in promoter\\n'\n",
      " 'Some promoter entry clones contain ATGs (pCG142, pCM1.41, Vidal promoters), '\n",
      " 'while others do not (pCM1.127, pCM1.128). If the\\n'\n",
      " 'ATG is included with the promoter in the first position clone, and the ORF '\n",
      " 'is in the middle position, then 1) the ORF must\\n'\n",
      " \"be cloned so as to be in frame (this is typically done by adding 2 N's to \"\n",
      " \"the 5' end) and 2) the ORF will be extended at its\\n\"\n",
      " \"5' end by 9 amino acids encoded by the attB site (Figure 4B).\\n\"\n",
      " '4.5.2.\\xa0- STOP codon in ORF\\n'\n",
      " \"If the STOP codon is included in the ORF (middle position) and the 3'UTR is \"\n",
      " \"in the third position, then the 3'UTR will be\\n\"\n",
      " \"extended at its 5' end by the attB site sequence. In our experience (Merritt \"\n",
      " 'et al., 2008), this is an acceptable compromise that does not appear to '\n",
      " 'create problems.\\n'\n",
      " '4.5.3.\\xa0- STOP codon NOT in ORF\\n'\n",
      " 'Some ORF clones (Vidal collection) do not contain STOP codons, in which case '\n",
      " \"the STOP codon must be included with the 3'UTR.\\n\"\n",
      " \"If the ORF is in the Middle position and the STOP+3'UTR are in the third \"\n",
      " 'position, the transgenic protein will be extended\\n'\n",
      " \"at its 3' end by 9 amino acids encoded by the attB site.\\n\"\n",
      " '4.6.\\xa0Other linker issue\\n'\n",
      " 'We have also found that placing the attB1 linker between GFP and ORF '\n",
      " 'sequences can interfere with chromatin localization in germline pachytene '\n",
      " 'nuclei. We have not\\n'\n",
      " 'observed this problem with the attB2 linker (Figure 5).\\n'\n",
      " 'Figure 5: Inclusion of attB1 linker between GFP and ORF can affect '\n",
      " \"distribution of fusion protein.\\xa0The pie-1 promoter and tbb-2 3'UTR drive \"\n",
      " 'expression in all germ cells. However depending on the configuration of the '\n",
      " 'attB sites, the fusion protein will behave differently. If the attB1 site is '\n",
      " 'placed between GFP and H2B, the resulting fusion protein is unstable in '\n",
      " 'pachytene nuclei (B). The attB2 site, however, does not cause this problem '\n",
      " '(A).\\n'\n",
      " '5.\\xa0Choosing a destination vector\\n'\n",
      " 'Choice of destination vector will depend on the choice of transformation '\n",
      " 'method to be used. There are currently two recommended\\n'\n",
      " 'transformation methods for germline transgenes: bombardment and '\n",
      " 'Mos1-mediated Single Copy Insertion (MosSCI).\\n'\n",
      " 'Bombardment does not require injection, but requires many worms '\n",
      " '(approximately 200,000 adults per construct) and is relatively\\n'\n",
      " 'slow (∼1 month), although hands-on time is low. Insertions are random '\n",
      " 'throughout the genome. Expression levels are remarkably\\n'\n",
      " 'similar from line to line, and generally lower than endogenous levels '\n",
      " '(Seydoux lab, unpublished; Tenlen et al., 2008). Bombardment uses '\n",
      " 'Destination Vector pCG150, which contains the unc-119 minimal rescue '\n",
      " 'fragment. See the bombardment protocol below for directions on how to use '\n",
      " 'this marker effectively to obtain\\n'\n",
      " 'low copy insertions.\\n'\n",
      " 'Mos1-mediated Single Copy Insertion (MosSCI) yields single-copy transgenes '\n",
      " 'inserted at a pre-selected site. This method uses\\n'\n",
      " 'the standard injection method and does not require a gene gun. Stable lines '\n",
      " 'can be obtained in less than 2 weeks. MosSCI uses\\n'\n",
      " 'Destination Vector PCFJ150. See Frokjaer-Jensen et al., 2008 and '\n",
      " 'http://sites.google.com/site/jorgensenmossci/Home for protocols.\\n'\n",
      " 'Destination Vectors MUST BE grown in DB3.1 bacteria. DB3.1 bacteria can '\n",
      " 'carry plasmids with the ccdB “death gene” that is used as a negative '\n",
      " 'selectable marker for Gateway® cloning. Destination vectors should be '\n",
      " 'selected on ampicillin (Amp) and chloramphenicol (Cm).\\n'\n",
      " 'Table 3: Selected Destination Vectors\\n'\n",
      " 'Destination Vector (reference)\\n'\n",
      " 'Selectable Marker\\n'\n",
      " 'Description\\n'\n",
      " 'Distributed by\\n'\n",
      " 'pDESTR4-R3\\n'\n",
      " 'none\\n'\n",
      " 'original Destination Vector for Multisite Gateway® cloning\\n'\n",
      " 'Invitrogen\\n'\n",
      " 'pCG150 (Merritt et al., 2008)\\n'\n",
      " 'unc-119\\n'\n",
      " 'pDESTR4-R3 Destination Vector with a 2.2 kb C. elegans unc-119 rescuing '\n",
      " 'fragment (with cDNA as ORF) cloned into the vector backbone.\\n'\n",
      " 'Addgene (http://www.addgene.org/pgvec1)\\n'\n",
      " 'pCFJ150 (Frokjaer-Jensen et al., 2008)\\n'\n",
      " 'unc-119\\n'\n",
      " 'MosSCI Destination Vector. Contains 2.1 kb C. briggsae unc-119 rescuing '\n",
      " 'fragment and Multisite Gateway® cassette flanked by Mos1 site sequences for '\n",
      " 'insertion at Mos1 site by homologous recombination.\\n'\n",
      " 'Addgene (http://www.addgene.org/pgvec1)\\n'\n",
      " '6.\\xa0Non-Multisite Gateway® vectors\\n'\n",
      " '6.1.\\xa0Traditional cloning vectors\\n'\n",
      " 'Table 4 shows commonly used vectors that use restriction enzyme cloning '\n",
      " \"methods. These vectors contain pie-1 promoter and pie-1 3' sequences, and \"\n",
      " 'therefore are best suited for expression in oocytes and early embryos '\n",
      " '(maternal expression).\\n'\n",
      " 'Table 4: Traditional cloning vectors for embryonic expression\\n'\n",
      " 'Vector (reference)\\n'\n",
      " 'Selectable Marker\\n'\n",
      " 'Vector components\\n'\n",
      " 'Description\\n'\n",
      " 'Distributed by\\n'\n",
      " 'pJH4.52 (Strome et al., 2001)\\n'\n",
      " 'none\\n'\n",
      " \"pie-1 promoter::GFP::(SpeI) H2B (SpeI)::pie-1 3'\\n\"\n",
      " 'Replace Histone H2B with ORF of interest using flanking SpeI sites\\n'\n",
      " 'Geraldine Seydoux gseydoux@jhmi.edu\\n'\n",
      " 'pAZ132 (Praitis, et al 2001)\\n'\n",
      " 'unc-119\\n'\n",
      " \"pie-1 promoter::GFP::(SpeI) H2B (SpeI)::pie-1 3'\\n\"\n",
      " 'Replace Histone H2B with ORF of interest using flanking SpeI sites\\n'\n",
      " 'Vida Praitis praitis@grinnell.edu\\n'\n",
      " 'pIC26 (http://jura.wi.mit.edu/cheeseman/Plasmids.php)\\n'\n",
      " 'unc-119\\n'\n",
      " \"pie-1 promoter::GFP::TEV::S-peptide::(SpeI)::pie-1 3'\\n\"\n",
      " 'Worm LAP construct (with GFP), clone ORF of interest into SpeI site\\n'\n",
      " 'Iain Cheeseman icheese@wi.mit.edu\\n'\n",
      " 'pAA65 (http://jura.wi.mit.edu/cheeseman/Plasmids.php)\\n'\n",
      " 'unc-119\\n'\n",
      " \"pie-1 promoter::mCherry::TEV::S-peptide::(SpeI)::pie-1 3'\\n\"\n",
      " 'Worm LAP construct (with mCherry), clone ORF of interest into SpeI site\\n'\n",
      " 'Iain Cheeseman icheese@wi.mit.edu\\n'\n",
      " \"pSO26 (O'Rourke et al., 2007)\\n\"\n",
      " 'unc-119\\n'\n",
      " \"pie-1 promoter::GFP::TEV::S-peptide::MCS(6)::pie-1 3'\\n\"\n",
      " 'Worm LAP construct (with GFP) derived from pIC26 (above), contains 6 '\n",
      " 'restriction sites (MCS) in place of SpeI.\\n'\n",
      " 'Bruce Bowerman bbowerman@molbio.uoregon.edu\\n'\n",
      " 'Other vectors only include a multiple cloning site and the transformation '\n",
      " 'marker:\\n'\n",
      " 'pDP#MM016b contains large genomic copy of the unc-119-rescuing fragment and '\n",
      " 'pDP#MM051 contains a smaller copy of the unc-119-rescuing fragment (cDNA '\n",
      " 'sequences instead of genomic sequences for the unc-119 ORF) (Maduro and '\n",
      " 'Pilgrim, 1995). In our hands, there is no difference in the rescuing ability '\n",
      " 'of these unc-119-rescuing fragments.\\n'\n",
      " 'pCFJ150 and pCFJ151 are two MCS-containing vectors that can be used for Mos1 '\n",
      " 'mediated Single Copy Insertion (MosSCI) (Frokjaer-Jensen et al., 2008). '\n",
      " 'These vectors contain the small C. briggsae unc-119 rescuing fragment and an '\n",
      " 'MCS flanked by Mos1 site sequences for insertion at Mos1 site by homologous '\n",
      " 'recombination.\\n'\n",
      " '6.2.\\xa0Single site Gateway® vectors\\n'\n",
      " 'Table 5 shows some commonly used single site Gateway® vectors for embryonic '\n",
      " 'expression. These vectors contain a Gateway® cassette that is capable of '\n",
      " 'accepting ORF sequences cloned into pDONR201/221 (equivalent to the middle '\n",
      " 'Entry Clone in the\\n'\n",
      " 'Multisite Gateway® system) These vectors all contain pie-1 promoter and '\n",
      " \"pie-1 3' sequences to drive expression in oocytes and embryos.\\n\"\n",
      " 'Table 5\\n'\n",
      " 'Vector (reference)\\n'\n",
      " 'Selectable Marker\\n'\n",
      " 'Vector components\\n'\n",
      " 'Description\\n'\n",
      " 'Distributed by\\n'\n",
      " 'pKR2.40 (Reese et al., 2000)\\n'\n",
      " 'none\\n'\n",
      " \"pie-1 promoter::GFP::[Gateway® Cassette]::pie-1 3'\\n\"\n",
      " 'To GFP tag ORF, does not contain unc-119 rescuing fragment\\n'\n",
      " 'Geraldine Seydoux gseydoux@jhmi.edu\\n'\n",
      " \"pID2.02 (D'Agostino et al., 2006)\\n\"\n",
      " 'unc-119\\n'\n",
      " \"pie-1 promoter::[Gateway® Cassette]::pie-1 3'\\n\"\n",
      " 'To drive ORF, ORF must be fused with tag by researcher if desired\\n'\n",
      " 'Geraldine Seydoux gseydoux@jhmi.edu\\n'\n",
      " \"pID3.01 (D'Agostino et al., 2006)\\n\"\n",
      " 'unc-119\\n'\n",
      " \"pie-1 promoter::GFP::[Gateway® Cassette]::pie-1 3'\\n\"\n",
      " 'To GFP tag ORF\\n'\n",
      " 'Geraldine Seydoux gseydoux@jhmi.edu\\n'\n",
      " 'pCM2.03 (Merritt et al., 2008)\\n'\n",
      " 'unc-119\\n'\n",
      " 'pie-1 promoter::GFP::[Gateway® Cassette]\\n'\n",
      " \"To GFP tag ORF under the control of any 3'UTR. 3'UTR sequences must also be \"\n",
      " 'included downstream of ORF.\\n'\n",
      " 'Geraldine Seydoux gseydoux@jhmi.edu\\n'\n",
      " '7.\\xa0Protocols\\n'\n",
      " '7.1.\\xa0Gateway® cloning\\n'\n",
      " \"The following protocols are modified from Invitrogen's protocols. Original \"\n",
      " \"protocols can be found on Invitrogen's website\\n\"\n",
      " '(http://www.invitrogen.com/site/us/en/home.html).\\n'\n",
      " '7.2.\\xa0Oligo design/PCR\\n'\n",
      " 'To clone an insert into an Entry Vector, PCR primers must contain the '\n",
      " 'flanking attB sites:\\n'\n",
      " \"For pDONRP4-P1R (5' Entry Clones) add:\\n\"\n",
      " \"Forward - attB4: 5'-GGGG ACA ACT TTG TAT AGA AAA GTT GNN\\n\"\n",
      " \"Reverse - attB1r: 5'-GGGG AC TGC TTT TTT GTA CAA ACT TGN\\n\"\n",
      " \"(where N's put coding sequences in frame)\\n\"\n",
      " 'For pDONR201/221 (Middle Entry Clones) add:\\n'\n",
      " \"Forward - attB1: 5'-GGGG ACA AGT TTG TAC AAA AAA GCA GGC TNN\\n\"\n",
      " \"Reverse - attB2: 5'-GGGG AC CAC TTT GTA CAA GAA AGC TGG GTN\\n\"\n",
      " \"(where N's put coding sequences in frame)\\n\"\n",
      " \"For pDONRP2R-P3 (3' Entry Clones) add:\\n\"\n",
      " \"Forward - attB2r: 5'-GGGG ACA GCT TTC TTG TAC AAA GTG GNN\\n\"\n",
      " 'Note: The MultiSite Gateway® Three-Fragment Vector Construction Kit Manual, '\n",
      " 'Version E, 16 August 2007 has the INCORRECT sequence of this oligo: GGGG '\n",
      " 'ACA\\n'\n",
      " 'GCT TTC TTG TAT AGA AAA GTT GNN instead of the CORRECT sequence (above) that '\n",
      " 'can be found in the manual Version G, 8 September\\n'\n",
      " '2008.\\n'\n",
      " \"Reverse- attB3: 5'-GGGG AC AAC TTT GTA TAA TAA AGT TGN\\n\"\n",
      " \"(where N's put coding sequences in frame)\\n\"\n",
      " 'Some examples:\\n'\n",
      " 'To clone a promoter with its ATG into pDONRP4-P1R:\\n'\n",
      " \"Forward oligo - includes attB4 (sense) :: 5' end of promoter sequence \"\n",
      " '(sense):\\n'\n",
      " \"5'-GGGG ACA ACT TTG TAT AGA AAA GTT G NNNNNN-3'\\n\"\n",
      " \"Reverse oligo—includes attB1r (antisense) :: 3' end of promoter sequence \"\n",
      " '(antisense):\\n'\n",
      " \"5'-GGGG AC TGC TTT TTT GTA CAA ACT TGT CAT NNNNNN-3'\\n\"\n",
      " '(where CAT is the start codon (antisense of ATG), and T puts attB1r '\n",
      " 'sequences in frame)\\n'\n",
      " 'To clone a promoter without its ATG into pDONRP4-P1R:\\n'\n",
      " \"Forward oligo - includes attB4 (sense) :: 5' end of promoter sequence \"\n",
      " '(sense):\\n'\n",
      " \"5'-GGGG ACA ACT TTG TAT AGA AAA GTT G NNNNNN-3'\\n\"\n",
      " \"Reverse oligo—includes attB1r (antisense) :: 3' end of promoter sequence \"\n",
      " '(antisense):\\n'\n",
      " \"5'-GGGG AC TGC TTT TTT GTA CAA ACT TGAT NNNNNN-3'\\n\"\n",
      " '(where AT is the first two bases of the start codon (antisense))\\n'\n",
      " 'To clone an ORF (incuding ATG and STOP codons) into pDONR201/221:\\n'\n",
      " \"Forward oligo - includes attB1 (sense) :: 5' end of ORF sequence (sense):\\n\"\n",
      " \"5'-GGGG ACA AGT TTG TAC AAA AAA GCA GGC TCA ATG NNN-3'\\n\"\n",
      " '(where CA puts attB1 sequences in frame, and ATG is the start codon)\\n'\n",
      " 'Note: We now clone all of ORFs with an additional AAA (Lys) codon upstream '\n",
      " 'of the ATG. The oligos are designed in this manner\\n'\n",
      " \"(where AAA is the additional codon): 5'-GGGG ACA AGT TTG TAC AAA AAA GCA GGC \"\n",
      " \"TCA AAA ATG NNN-3'\\n\"\n",
      " \"This is useful when using a promoter construct that doesn't include an ATG \"\n",
      " \"(see Figure 3C) as the stretch of A's helps in ATG selection (Marc Perry, \"\n",
      " 'personal communication). This construct can also be used with a promoter '\n",
      " 'construct that has an ATG.\\n'\n",
      " \"Reverse oligo—includes attB2(antisense) :: 3' end of ORF sequence \"\n",
      " '(antisense):\\n'\n",
      " \"5'-GGGG AC CAC TTT GTA CAA GAA AGC TGG GT TTA NNN-3'\\n\"\n",
      " '(where TTA is the stop codon (antisense))\\n'\n",
      " \"To clone a 3'UTR in to pDONRP2R-P3:\\n\"\n",
      " \"Forward oligo - includes attB2r (sense) :: 5' end of 3'UTR sequence \"\n",
      " '(sense):\\n'\n",
      " \"5'-GGGG ACA GCT TTC TTG TAC AAA GTG GGA TAA NNN-3'\\n\"\n",
      " '(where GA puts the TAA stop codon in frame, creating a GGA (Ala) codon)\\n'\n",
      " 'Note: You need not include the stop codon if the ORF has one already. If the '\n",
      " 'ORF has no stop codon (as in the Vidal ORFeome)\\n'\n",
      " \"including the STOP in the 3'UTR will add the attB2R sequence to your ORF.\\n\"\n",
      " \"Reverse oligo—includes attB3 (antisense) :: 3' end of 3'UTR sequence \"\n",
      " '(antisense):\\n'\n",
      " \"5'-GGGG AC AAC TTT GTA TAA TAA AGT TG NNNNNN-3'\\n\"\n",
      " 'A few notes about PCR:\\n'\n",
      " 'We typically PCR amplify with a high-fidelity polymerase (i.e. Phusion High '\n",
      " 'Fidelity DNA Polymerase, Finnzymes). The template can be from any source '\n",
      " '(genomic DNA, cDNA, plasmid DNA, etc.).\\n'\n",
      " 'We typically perform a 50 μl PCR reaction and gel purify half of the '\n",
      " 'reaction with the QIAquick Gel Extraction Kit (Qiagen).\\n'\n",
      " 'We typically elute the DNA from the column with 30 μl of H2O and use 1-7 μl '\n",
      " 'of this purified PCR for the BP reaction.\\n'\n",
      " '7.3.\\xa0BP Reaction\\n'\n",
      " 'For each reaction mix at RT in 1.5 mL tube:\\n'\n",
      " 'gel purified PCR in H2O (∼15-150 ng, or 1-7 μl of purified PCR - see above, '\n",
      " 'add more PCR product for larger PCR inserts)\\n'\n",
      " 'pDONR Vector (∼150 ng/μl) (typically add 1 μl of miniprepped pDONR vector)\\n'\n",
      " 'TE (pH 8) to 8 μl\\n'\n",
      " 'Thaw BP Clonase II enzyme mix on ice.\\n'\n",
      " 'Note: BP Clonase II enzyme mix includes the buffer!, the original BP Clonase '\n",
      " 'enzyme mix (without the “II” in the name) did\\n'\n",
      " 'not contain the buffer and was added separately.\\n'\n",
      " 'Add 2 μl of BP Clonase II enzyme mix (immediately return enzyme mix to '\n",
      " 'freezer). Mix by vortexing, then microcentrifuge briefly.\\n'\n",
      " 'Incubate at 25°C for 1 hour or overnight (we typically do overnight '\n",
      " 'incubations for larger inserts or if there is very little\\n'\n",
      " 'PCR product)\\n'\n",
      " 'Add 1 μl of the Proteinase K solution. Mix by vortexing, then '\n",
      " 'microcentrifuge briefly.\\n'\n",
      " 'Incubate samples at 37°C for 10 minutes.\\n'\n",
      " '7.4.\\xa0BP Reaction Transformation\\n'\n",
      " 'Transform 2 μl of each BP reaction into 50 μl of Max Efficiency or Library '\n",
      " 'Efficiency DH5α cells (DH5α cells with a transformation\\n'\n",
      " 'efficiency of 108) (Subcloning Efficiency DH5α cells can be used for highly '\n",
      " 'efficient BP reactions, i.e. very small inserts).\\n'\n",
      " 'Note: DH5α cells MUST be used for transformation. The success of the BP '\n",
      " 'reaction depends on negative selection against the ccdB gene\\n'\n",
      " 'to select against “empty” pDONR vectors. Not all E. coli strains are '\n",
      " 'sensitive to ccdB, for example DB3.1 the strain used\\n'\n",
      " 'to grow pDONR Vector, can tolerate ccdB. We have seen that XL1 Blue cells '\n",
      " 'also tolerate ccdB.\\n'\n",
      " 'Incubate on ice for 30 minutes.\\n'\n",
      " 'Heat-shock cells at 42°C for 30 seconds.\\n'\n",
      " 'Incubate on ice for 2 minutes.\\n'\n",
      " 'Add 280 μl of S.O.C. Medium (2% tryptone, 0.5% yeast extract, 10 mM sodium '\n",
      " 'chloride, 2.5 mM potassium chloride, 10 mM magnesium\\n'\n",
      " 'chloride, 10 mM magnesium sulfate, 20 mM glucose) and shake at 37°C for 1 '\n",
      " 'hour.\\n'\n",
      " 'Note: use of S.O.C (available from Invitrogen), instead of LB, will result '\n",
      " 'in many more colonies.\\n'\n",
      " 'Plate 30 μl and 300 μl of each transformation onto Kanamycin plates (50 '\n",
      " 'μg/ml) (NOT ampicillin!!)\\n'\n",
      " 'Incubate at 37°C overnight.\\n'\n",
      " 'Miniprep 2-4 colonies and roughly verify proper insert by restriction '\n",
      " 'digest, PCR, or running of uncut DNA on gel (Figure 6). Confirm candidate '\n",
      " 'preps by sequencing.\\n'\n",
      " 'Figure 6. Troubleshooting the BP reaction when using large or low-yield PCR '\n",
      " 'products.\\n'\n",
      " 'A. Most BP reactions yield the desired Entry clone with insert. Empty pDONR '\n",
      " 'plasmids resulting from an internal recombination\\n'\n",
      " 'reaction are obtained at a low frequency (C.M. unpublished) and are usually '\n",
      " 'rare enough not to be a problem. Empty pDONRs\\n'\n",
      " 'can, however, become a nuisance (be more common) if the BP is performed with '\n",
      " 'very small amounts of PCR insert or with a very\\n'\n",
      " 'long PCR insert (∼6 kb or greater).\\n'\n",
      " 'B. To help screen out empty pDONRs, DNA from a transformed colony is run '\n",
      " 'directly on a gel to verify that the approximate\\n'\n",
      " 'size plasmid is present (for protocol see: “Rapid Screening by Direct '\n",
      " 'Electrophoresis” section in Chapter 19 of E. coli Plasmid Vectors: Methods '\n",
      " 'and Applications). Minipreps are then performed only on those colonies that '\n",
      " 'contain the larger super-coiled plasmids. In this specific example,\\n'\n",
      " 'the BP reaction was done with a very low quantity (<15 ng) of a 4 kb PCR '\n",
      " 'product. The colony on the far left contains the\\n'\n",
      " 'pDONR vector with the 4kb insert, while the next 6 colonies are all empty '\n",
      " 'pDONRs.\\n'\n",
      " '7.5.\\xa0Multisite LR Reaction\\n'\n",
      " 'For each reaction mix at RT in 1.5 ml tube:\\n'\n",
      " '3 Entry Clones (∼10 fmoles each, typically just add 0.5 μl of miniprep)\\n'\n",
      " 'Destination Vector (∼20 fmoles, typically just add 1 μl of miniprep)\\n'\n",
      " 'TE (pH 8) to 8 μl\\n'\n",
      " 'Note: We typically do a negative control also - we remove one of the three '\n",
      " 'Entry Clones from the reaction (and get ∼0 colonies)\\n'\n",
      " 'Thaw LR Clonase II Plus enzyme mix on ice.\\n'\n",
      " 'Add 2 μl of LR Clonase II Plus enzyme mix (immediately return enzyme mix to '\n",
      " 'freezer). Mix by vortexing, then microcentrifuge\\n'\n",
      " 'briefly.\\n'\n",
      " 'Incubate at 25°C overnight.\\n'\n",
      " 'Add 1 μl of the Proteinase K solution. Mix by vortexing, then '\n",
      " 'microcentrifuge briefly.\\n'\n",
      " 'Incubate samples at 37°C for 10 minutes.\\n'\n",
      " '7.6.\\xa0Multisite LR Reaction Transformation\\n'\n",
      " 'Transform 2 μl of LR as described above for BP reaction BUT select on '\n",
      " 'ampicillin (100 μg/ml) instead of Kanamycin!!\\n'\n",
      " 'Miniprep 2-4 colonies and verify proper construct by restriction digest.\\n'\n",
      " '7.7.\\xa0Bombardment Protocol\\n'\n",
      " 'The bombardment technique can produce stable, low copy transgenes (1-3 '\n",
      " 'copies, Praitis et al., 2001) that will not be silenced in the germline. '\n",
      " 'Identification of transformed worms by bombardment is typically done by '\n",
      " 'rescuing\\n'\n",
      " 'unc-119 mutants with a wild-type copy of unc-119 present on the bombardment '\n",
      " 'plasmid. Bombardment has a very low frequency of successful transformation '\n",
      " 'per adult bombarded,\\n'\n",
      " 'so a large number of worms must be bombarded (approximately 200,000 adults '\n",
      " 'per construct). To select for transformed worms,\\n'\n",
      " 'worms are plated on several plates after bombardment and allowed to starve '\n",
      " 'for several days. Over this time, the unc-119-rescued worms have a selective '\n",
      " 'advantage over the non-transformed unc-119 worms, which cannot form dauers '\n",
      " 'and have difficulty surviving starvation (Praitis et al., 2001).\\n'\n",
      " 'Several bombardment protocols have been published elsewhere (examples: Green '\n",
      " 'et al., 2008; Praitis et al., 2001). The following protocol was created by '\n",
      " 'the Seydoux lab and is adapted from Shai Shaham, personal communication, and '\n",
      " 'Praitis et al., 2001. This protocol is for bombarding 3 different '\n",
      " 'constructs.\\n'\n",
      " 'Reagents:\\n'\n",
      " 'Biolistic PDS-1000/He particle delivery system #165-2257 Bio-Rad\\n'\n",
      " 'Hepta adapter, biolistic shock wave splitter #165-2225 Bio-Rad\\n'\n",
      " 'Biolistic Microcarriers #165-2335 Bio-Rad\\n'\n",
      " '1 μM gold beads #165-2263 Bio-Rad\\n'\n",
      " 'Rupture discs #165-2333 (1550psi) Bio-Rad\\n'\n",
      " 'Hepta stopping screen #165-2226 Bio-Rad\\n'\n",
      " 'Spermidine (tissue culture grade) S-4139 (5g) Sigma-Aldrich®\\n'\n",
      " 'Nystatin N1638 (100 mls) Sigma-Aldrich®\\n'\n",
      " 'You will need ∼160 large EP with Nystatin (NEP) plates spread with NA22 '\n",
      " 'bacteria to bombard 3 constructs.\\n'\n",
      " 'Enriched Peptone Plates with Nystatin (NEP) Recipe (1 Liter):\\n'\n",
      " '1.2 g sodium chloride\\n'\n",
      " '20 g peptone\\n'\n",
      " '25 g agar\\n'\n",
      " 'H2O to 1 liter\\n'\n",
      " 'autoclave- cool to 55°C- then add STERILE:\\n'\n",
      " '1 ml cholesterol (5 mg/ml in EtOH)\\n'\n",
      " '1 ml 1 M MgSO4\\n'\n",
      " '25 ml 1 M potassium phosphate (pH 6.0)\\n'\n",
      " '10 ml Nystatin suspension (10,000 units/ml)\\n'\n",
      " '7.7.1.\\xa0Many days prior to bombardment: Growing up worms\\n'\n",
      " '7.7.1.1.\\xa0Master plates\\n'\n",
      " 'Starting from an almost starved unc-119 (ed3 or 4) plate, take ∼10 piles of '\n",
      " 'adult unc-119 worms (unc-119 worms form piles when starved) and place onto '\n",
      " 'one small NGM plate with OP50 (see Maintenance of C. elegans, for NGM plate '\n",
      " 'recipe). Repeat for one other NGM plate for a total of two plates. Let the '\n",
      " 'worms grow at 25°C until food is\\n'\n",
      " 'gone and worms form piles.\\n'\n",
      " '7.7.1.2.\\xa0Amplify worms to 6 plates\\n'\n",
      " 'Resuspend the 2 plates in 6 ml M9 buffer total. Spread 1 ml of supernatant '\n",
      " 'to one large NEP plates with NA22 bacteria spread\\n'\n",
      " 'evenly over the entire plate. Repeat for a total of 6 plates. Let the worms '\n",
      " 'grow at 25°C to starved L1s or nearly starved\\n'\n",
      " 'adults (see below).\\n'\n",
      " 'Note: Worms can be grown at any temperature throughout the protocol (except '\n",
      " 'after bombardment, where we recommend 25°C to encourage GFP expression). '\n",
      " 'Grow at temperatures that are most convenient for you (from 15-25°C). The '\n",
      " 'scheme above for\\n'\n",
      " 'initially growing up the worms does not have to be followed strictly. In the '\n",
      " 'end, you just need several large plates (6 with\\n'\n",
      " 'newly starved L1s, or 12 with gravid adults).\\n'\n",
      " '7.7.2.\\xa02-4 days prior to bombardment: Final amplification of worms to 60 '\n",
      " 'plates\\n'\n",
      " 'The goal of the final amplification is to seed 60 plates with synchronized '\n",
      " 'L1s (approx 20,000 L1s/plate) and grow them to\\n'\n",
      " 'fairly synchronized adults that can be bombarded.\\n'\n",
      " '7.7.2.1.\\xa0If starting with starved L1s:\\n'\n",
      " 'Resuspend each plate of freshly-starved L1 in 10 ml of M9 buffer (total 60 '\n",
      " 'ml). Plate 1 ml of worm suspension onto a new large\\n'\n",
      " 'plate (60 total plates). Incubate the plates of L1s until they become young '\n",
      " 'adults.\\n'\n",
      " 'Note: USE ONLY FRESHLY-STARVED L1s. Plates that are over-starved will '\n",
      " 'contain dead L1s and will yield fewer worms for bombardment.\\n'\n",
      " 'Drop a little M9 on the worm piles to make sure they contain healthy '\n",
      " '(wiggly) L1s.\\n'\n",
      " 'If the plates contain many older worms as well as L1s, you must enrich for '\n",
      " 'L1s by washing all of the plates off with M9 into\\n'\n",
      " 'a 50 mL conical tube and letting the worms sit on the bench for about a '\n",
      " 'minute, letting the large worms sink to the bottom.\\n'\n",
      " 'Pipet the supernatant (enriched with L1s) into a new 50mL conical and use '\n",
      " 'these to spread to 60 plates.\\n'\n",
      " '7.7.2.2.\\xa0If starting with nearly starved (gravid) adults:\\n'\n",
      " 'Collect adults in M9, precipitate and resuspend in bleach mixture (1 part '\n",
      " '10N NaOH, 1 part Chlorox bleach, 8 parts M9). Rock\\n'\n",
      " 'worms for 7 minutes, spin, wash in M9. Amplify each plate of bleached eggs '\n",
      " 'to 10 plates (total of 60 plates). Incubate the\\n'\n",
      " 'plates until worms become young adults.\\n'\n",
      " 'Note: For best synchronization, eggs can be hatched overnight in M9 before '\n",
      " 'plating on the 60 plates. If worms are hatched\\n'\n",
      " 'in M9 overnight, it is easy to count L1s. Ideally, ∼20,000 L1 should be '\n",
      " 'plated per large NEP plate. With this number of worms,\\n'\n",
      " 'the plates will not starve completely before the worms reach adulthood, but '\n",
      " 'will have low enough food to facilitate washing\\n'\n",
      " 'off the worms. Typically, 6 plates of adults with eggs are sufficient to '\n",
      " 'collect enough L1s for 60 plates. To be safe, we\\n'\n",
      " 'often bleach more that 6 plates (12+ plates) to ensure that we obtain enough '\n",
      " 'L1s for bombardment.\\n'\n",
      " '7.7.3.\\xa0Bombardment Day:\\n'\n",
      " 'Autoclave Hepta adaptor, microcarrier holder, microcarriers, mesh stopping '\n",
      " 'screens and forceps.\\n'\n",
      " '7.7.3.1.\\xa0Worm preparation\\n'\n",
      " 'Make sure most worms are YOUNG ADULTS (hermaphrodites with single row of '\n",
      " 'embryos).\\n'\n",
      " 'Wash worms from the 60 plates with M9 into a 50 ml tube. WATCH FOR '\n",
      " 'CONTAMINATION. Spin at low speed for 1 minute at room temperature;\\n'\n",
      " 'wash worms until M9 solution becomes clear and finally transfer worms to a '\n",
      " '15 ml tube. Spin at low speed, you should end up\\n'\n",
      " 'with a 3-4 ml pellet of packed adult worms. Remove liquid and resuspend '\n",
      " 'worms in M9 to a total volume of 12 ml.\\n'\n",
      " 'Using a Pasteur pipette, spread worms onto the surface of a dry enriched '\n",
      " 'peptone plate with NA22. Add them drop-wise starting\\n'\n",
      " 'at the center and then spiraling around until you reach the edge of the '\n",
      " 'plate. Repeat until all worms are plated (6 plates).\\n'\n",
      " 'Leave the covers off the plates to evaporate the liquid. This should take no '\n",
      " 'more than 15 min.\\n'\n",
      " 'While the plates dry, prepare the gene gun. We use the Bio-Rad Biolistic '\n",
      " 'PDS-1000/He particle delivery system with Hepta adaptor.\\n'\n",
      " 'This adaptor saves an enormous amount of time and effort. You can find the '\n",
      " 'illustrations and definitions of the various jargon\\n'\n",
      " 'terms (microcarriers and microcarrier holder) in the Bio-Rad manual. Read '\n",
      " 'this manual to be familiar with the procedures described\\n'\n",
      " 'below.\\n'\n",
      " 'Wipe down the bombardment chamber and chamber door with 70% EtOH. Unscrew '\n",
      " 'the helium pressure gauge and immerse in isopropanol\\n'\n",
      " 'for 15 min. or longer. Screw back the helium pressure gauge.\\n'\n",
      " '7.7.3.2.\\xa0DNA preparation\\n'\n",
      " 'Weigh 35—50 mg of 1 μM gold beads (Bio-Rad) into siliconized 1.5 ml '\n",
      " 'eppendorf tube.\\n'\n",
      " 'Note: tungsten beads have also been used successfully (Jadhav et al., 2008)\\n'\n",
      " 'Add 1 ml 70% EtOH. Vortex 5min. Soak for 15 min. Pellet and remove '\n",
      " 'supernatant\\n'\n",
      " 'Add 1 ml sterile water. Vortex 1min. Soak for 1 min. Pellet and remove '\n",
      " 'supernatant\\n'\n",
      " 'Add 1 ml sterile water. Vortex 1min. Soak for 1 min. Pellet and remove '\n",
      " 'supernatant\\n'\n",
      " 'Add 1 ml sterile water. Vortex 1min. Soak for 1 min. Pellet and remove '\n",
      " 'supernatant\\n'\n",
      " 'Resuspend in 500 μl sterile 50% glycerol. This bead stock can be used for 2 '\n",
      " 'weeks (or more and should be stored at 4°C).\\n'\n",
      " 'Vortex mix for 5min. Remove immediately 100 μl of bead suspension and place '\n",
      " 'into a siliconized eppendorf tube. Repeat twice\\n'\n",
      " 'for a total of 3 tubes. Make sure to keep the gold beads in suspension. One '\n",
      " 'tube will be used for each construct.\\n'\n",
      " 'To each tube, add in order while vortexing on medium speed:\\n'\n",
      " '10 μg DNA (typically, 30-50 μl of miniprep, or 10 μl of 1 μg/μl midi)\\n'\n",
      " '100 μl 2.5M CaCl2\\n'\n",
      " '40 μl 0.1M spermidine\\n'\n",
      " 'Vortex 2min. Soak for 1min. Pellet and remove supernatant.\\n'\n",
      " 'Add 280 μl 70%EtOH. Flick tube to mix. Pellet and remove supernatant.\\n'\n",
      " 'Add 280 μl 100%EtOH. Flick tube to mix. Pellet and remove sup. Add 100 μl '\n",
      " '100%EtOH and resuspend by gently flicking tube.\\n'\n",
      " 'This is your prepped DNA.\\n'\n",
      " '7.7.3.3.\\xa0Bombardment\\n'\n",
      " 'Open vacuum port, turn on gene gun. Open helium tank valve (check to make '\n",
      " 'sure the He tank pressure is >2200psi). Close door.\\n'\n",
      " 'Vortex your DNA preparation at medium speed with cap closed. Stop and '\n",
      " 'quickly transfer 7 μl of beads onto the middle of a\\n'\n",
      " 'microcarrier. Repeat for 14 microcarriers total. Let ethanol evaporate (this '\n",
      " 'only takes a few minutes).\\n'\n",
      " 'Place 7 microcarriers (with dried DNA-gold beads) onto the Hepta adaptor '\n",
      " 'microcarrier holder using forceps and tighten up\\n'\n",
      " 'with the special tool.\\n'\n",
      " 'Place a rupture disk soaked in isopropanol in the retaining cap and tighten. '\n",
      " 'Place stopping screen and microcarrier holder\\n'\n",
      " 'in chamber as described in the manual. Place uncovered worm plate (taped to '\n",
      " 'the sample holder using a rolled piece of adhesive\\n'\n",
      " 'tape to create double sided tape) onto second rung in bombardment chamber.\\n'\n",
      " 'Pull vacuum to 27 In. of Hg. Press Fire button until disk ruptures. Release '\n",
      " 'vacuum (Vent position), and remove plate.\\n'\n",
      " 'Repeat 1-4 for with the next 7 microcarriers and a new plate of worms.\\n'\n",
      " 'Repeat 1-5 for the next two DNA preparations (each construct is bombarded '\n",
      " 'onto two plates)\\n'\n",
      " 'Turn off vacuum. Close helium tank valve. Make sure no pressure is left in '\n",
      " 'the line. Turn the gene gun power OFF.\\n'\n",
      " '7.7.3.4.\\xa0Plating worms\\n'\n",
      " 'Resuspend each plate of bombarded worms in 13 ml of M9 (repeat for each of '\n",
      " 'the 6 plates). Plate 1 ml of worms onto each fresh\\n'\n",
      " 'NEP plate (approximately 8,000 bombarded worms per plate). For each '\n",
      " 'construct, you will have 26 plates or 78 plates for all\\n'\n",
      " 'three constructs.\\n'\n",
      " 'Note: To reduce contamination at this step, add 2.5 μg/ml Fungizone '\n",
      " '(amphotericin B) and 20 μg/ml tetracycline to the M9 to\\n'\n",
      " 'inhibit fungal and bacterial growth.\\n'\n",
      " 'Let plates dry on the bench and incubate at 25°C for ∼2 weeks. The worms '\n",
      " 'should starve no sooner than Day 3 (if kept at 25°C).\\n'\n",
      " 'If they starve earlier, you may not obtain stable integrants. It is very '\n",
      " 'important to plate the bombarded worms at a low enough\\n'\n",
      " 'density to avoid starvation in the first 3 days.\\n'\n",
      " '7.7.4.\\xa02 weeks after bombardment:\\n'\n",
      " 'Scan plates for wild-type worms.\\n'\n",
      " 'Note: You can continue to scan for wild-type worms for 2 additional weeks.\\n'\n",
      " 'From each plate with wild-type worms, pick 3-5 worms and transfer them to '\n",
      " 'one small NGM plate. Place the plates at 25°C for\\n'\n",
      " '4 days, and then examine the progeny for GFP expression. You can clone out '\n",
      " 'at this point (pick a single GFP positive worm\\n'\n",
      " 'from one plate to start the line). Make sure the lines are stable (do not '\n",
      " 'segregate Uncs). Lines that segregate Uncs tend\\n'\n",
      " 'to loose germline expression over time. Keep the lines at 25°C to maintain '\n",
      " 'GFP (not all lines need to be kept at 25°C but\\n'\n",
      " 'many do).\\n'\n",
      " 'In our hands, GFP expression in the germline is observed in 10-50% of '\n",
      " 'independent unc-119 rescued lines (lines are considered “independent” when '\n",
      " 'isolated from different plates). We typically bombard the same transgene\\n'\n",
      " 'twice, on different days, to increase the likelihood of getting a line that '\n",
      " 'expresses in the germline (transformation success\\n'\n",
      " 'can vary from day to day). If 20 or more unc-119 rescued lines (stable) are '\n",
      " 'obtained and none express in the germline, it may be helpful to bombard in '\n",
      " 'parallel a construct\\n'\n",
      " 'known to express well (examples: pCM1.34 (pie-1 promoter::GFP::histone '\n",
      " \"H2B::tbb-2 3'UTR) or pCM4.09 (pie-1 promoter::GFP::pie-1 ORF::pie-1 3'UTR \"\n",
      " '(Merritt et al., 2008) to verify the technique is working.\\n'\n",
      " 'Final note: In the end, there are several key steps to a successful '\n",
      " 'bombardment:\\n'\n",
      " 'Start with synchronized worms\\n'\n",
      " 'Bombard adults with no more than 1 row of eggs\\n'\n",
      " 'Bombard ∼0.5mL of packed adults per plate\\n'\n",
      " 'After bombardment, plate worms onto new plates so they starve in ∼3 days, '\n",
      " 'NOT EARLIER!\\n'\n",
      " 'Avoid contamination!!\\n'\n",
      " 'Keep lines at 25°C.\\n'\n",
      " '8.\\xa0References\\n'\n",
      " 'Chalfie, M., Tu, Y., Euskirchen, G., Ward, W., and Prasher, D.C. (1994). '\n",
      " 'Green fluorescent protein as a marker for gene expression.\\n'\n",
      " 'Science 263, 802-805.\\n'\n",
      " 'Abstract\\n'\n",
      " 'Article\\n'\n",
      " \"D'Agostino, I., Merritt, C., Chen, P.L., Seydoux, G., and Subramaniam, K. \"\n",
      " '(2006). Translational repression restricts expression\\n'\n",
      " 'of the C. elegans Nanos homolog NOS-2 to the embryonic germline. Dev. Biol. '\n",
      " '292, 244-252.\\n'\n",
      " 'Article\\n'\n",
      " 'Dupuy, D., Li, Q.R., Deplancke, B., Boxem, M., Hao, T., Lamesch, P., '\n",
      " 'Sequerra, R., Bosak, S., Doucette-Stamm, L., Hope, I.A., et al. (2004). A '\n",
      " 'first version of the Caenorhabditis elegans Promoterome. Genome Res. 14, '\n",
      " '2169-2175.\\n'\n",
      " 'Abstract\\n'\n",
      " 'Article\\n'\n",
      " 'Fire, A. (1986). Integrative transformation of C. elegans. EMBO J. 5, '\n",
      " '2673-2680.\\n'\n",
      " 'Abstract\\n'\n",
      " 'Frøkjaer-Jensen, C., Davis, M.W., Hopkins, C.E., Newman, B.J., Thummel, '\n",
      " 'J.M., Olesen, S.P., Grunnet, M., and Jorgensen, E.M.\\n'\n",
      " '(2008). Single-copy insertion of transgenes in Caenorhabditis elegans. Nat. '\n",
      " 'Genet. 40, 1375-1383.\\n'\n",
      " 'Abstract\\n'\n",
      " 'Article\\n'\n",
      " 'Green, R.A., Audhya, A., Pozniakovsky, A., Dammermann, A., Pemble, H., '\n",
      " 'Monen, J., Portier, N., Hyman, A., Desai, A., and Oegema,\\n'\n",
      " 'K. (2008). Expression and imaging of fluorescent proteins in the C. elegans '\n",
      " 'gonad and early embryo. Methods Cell Biol. 85, 179-218.\\n'\n",
      " 'Abstract\\n'\n",
      " 'Article\\n'\n",
      " 'Hubbard, E.J., and Greenstein, D. (2005). Introduction to the germ line. '\n",
      " 'WormBook, ed. The C. elegans Research Community, WormBook, '\n",
      " 'doi/10.1895/wormbook.1.18.1, http://www.wormbook.org.\\n'\n",
      " 'Abstract\\n'\n",
      " 'Article\\n'\n",
      " 'Hunt-Newbury, R., Viveiros, R., Johnsen, R., Mah, A., Anastas, D., Fang, L., '\n",
      " 'Halfnight, E., Lee, D., Lin, J., Lorch, A., et al. (2007). High-throughput '\n",
      " 'in vivo analysis of gene expression in Caenorhabditis elegans. PLoS Biol. 5, '\n",
      " 'e237.\\n'\n",
      " 'Abstract\\n'\n",
      " 'Article\\n'\n",
      " 'Jadhav, S., Rana, M., and Subramaniam, K. (2008). Multiple maternal proteins '\n",
      " 'coordinate to restrict the translation of C. elegans nanos-2 to primordial '\n",
      " 'germ cells. Development 135, 1803-1812.\\n'\n",
      " 'Article\\n'\n",
      " 'Kelly, W.G., Xu, S., Montgomery, M.K., and Fire, A. (1997). Distinct '\n",
      " 'requirements for somatic and germline expression of a\\n'\n",
      " 'generally expressed Caernorhabditis elegans gene. Genetics 146, 227-238.\\n'\n",
      " 'Abstract\\n'\n",
      " 'Lamesch, P., Milstein, S., Hao, T., Rosenberg, J., Li, N., Sequerra, R., '\n",
      " 'Bosak, S., Doucette-Stamm, L., Vandenhaute, J., Hill,\\n'\n",
      " 'D.E., et al. (2004). C. elegans ORFeome version 3.1: increasing the coverage '\n",
      " 'of ORFeome resources with improved gene predictions. Genome Res. 14, '\n",
      " '2064-2069.\\n'\n",
      " 'Abstract\\n'\n",
      " 'Article\\n'\n",
      " 'Maduro, M., and Pilgrim, D. (1995). Identification and cloning of unc-119, a '\n",
      " 'gene expressed in the Caenorhabditis elegans nervous system. Genetics 141, '\n",
      " '977-988.\\n'\n",
      " 'Abstract\\n'\n",
      " 'Mangone, M., Macmenamin, P., Zegar, C., Piano, F., and Gunsalus, K.C. '\n",
      " \"(2008). UTRome.org: a platform for 3'UTR biology in\\n\"\n",
      " 'C. elegans. Nucleic Acids Res. 36, D57-62.\\n'\n",
      " 'Abstract\\n'\n",
      " 'Article\\n'\n",
      " 'Mello, C., and Fire, A. (1995). DNA transformation. Methods Cell Biol. 48, '\n",
      " '451-482.\\n'\n",
      " 'Abstract\\n'\n",
      " 'Article\\n'\n",
      " \"Merritt, C., Rasoloson, D., Ko, D., and Seydoux, G. (2008). 3'UTRs are the \"\n",
      " 'primary regulators of gene expression in the C. elegans germline. Curr. '\n",
      " 'Biol. 18, 1476-1482.\\n'\n",
      " 'Abstract\\n'\n",
      " \"O'Rourke, S.M., Dorfman, M.D., Carter, J.C., and Bowerman, B. (2007). Dynein \"\n",
      " 'modifiers in C. elegans: light chains suppress conditional heavy chain '\n",
      " 'mutants. PLoS Genet. 3, e128.\\n'\n",
      " 'Article\\n'\n",
      " 'Praitis, V., Casey, E., Collar, D., and Austin, J. (2001). Creation of '\n",
      " 'low-copy integrated transgenic lines in Caenorhabditis elegans. Genetics '\n",
      " '157, 1217-1226.\\n'\n",
      " 'Abstract\\n'\n",
      " 'Reese, K.J., Dunn, M.A., Waddle, J.A., and Seydoux, G. (2000). Asymmetric '\n",
      " 'segregation of PIE-1 in C. elegans is mediated by two complementary '\n",
      " 'mechanisms that act through separate PIE-1 protein domains. Mol. Cell 6, '\n",
      " '445-455.\\n'\n",
      " 'Article\\n'\n",
      " 'Robert, V.J., Vastenhouw, N.L., and Plasterk, R.H. (2004). RNA interference, '\n",
      " 'transposon silencing, and cosuppression in the\\n'\n",
      " 'Caenorhabditis elegans germ line: similarities and differences. Cold Spring '\n",
      " 'Harb. Symp. Quant. Biol. 69, 397-402.\\n'\n",
      " 'Abstract\\n'\n",
      " 'Strome, S., Powers, J., Dunn, M., Reese, K., Malone, C.J., White, J., '\n",
      " 'Seydoux, G., and Saxton, W. (2001). Spindle dynamics\\n'\n",
      " 'and the role of gamma-tubulin in early Caenorhabditis elegans embryos. Mol. '\n",
      " 'Biol. Cell 12, 1751-1764.\\n'\n",
      " 'Tenlen, J.R., Molk, J.N., London, N., Page, B.D., and Priess, J.R. (2008). '\n",
      " 'MEX-5 asymmetry in one-cell C. elegans embryos requires PAR-4- and '\n",
      " 'PAR-1-dependent phosphorylation. Development 135, 3665-3675.\\n'\n",
      " 'Abstract\\n'\n",
      " 'Article\\n'\n",
      " 'Wilm, T., Demel, P., Koop, H.U., Schnabel, H., and Schnabel, R. (1999). '\n",
      " 'Ballistic transformation of Caenorhabditis elegans. Gene 229, 31-35.\\n'\n",
      " 'Abstract\\n'\n",
      " 'Article\\n'\n",
      " '*Edited by Oliver Hobert and Susan Mango. Last revised July 1, 2009. '\n",
      " 'Published February 8, 2010. This chapter should be cited\\n'\n",
      " 'as: Merritt, C., et al. Transgenic solutions for the germline (February 8, '\n",
      " '2010), WormBook, ed. The C. elegans Research Community, WormBook, '\n",
      " 'doi/10.1895/wormbook.1.148.1, http://www.wormbook.org.\\n'\n",
      " 'Copyright: © 2010 Christopher Merritt et al. This is an open-access article '\n",
      " 'distributed under the terms of the Creative\\n'\n",
      " 'Commons Attribution License, which permits unrestricted use, distribution, '\n",
      " 'and reproduction in any medium, provided the original\\n'\n",
      " 'author and source are credited.\\n'\n",
      " '§To whom correspondence should be addressed. E-mail: gseydoux@jhmi.edu\\n'\n",
      " 'All WormBook content, except where otherwise noted, is licensed under a '\n",
      " 'Creative Commons Attribution License.')\n"
     ]
    }
   ],
   "source": [
    "pp.pprint(print_best_text('Transgenicask0.html'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
